router/src/infer/v3/queue.rs

```diff
@@ -120,12 +135,14 @@ async fn queue_task(
                 token_budget,
                 response_sender,
                 span,
-            } => span.in_scope(|| {
-                let next_batch =
-                    state.next_batch(min_size, max_size, prefill_token_budget, token_budget);
+            } => {
+                let _parent_span = span.enter();
```

Comment:
`span.enter()` doesn't work properly in async context, use [instrument](https://docs.rs/tracing/latest/tracing/trait.Instrument.html) instead.

Reply:
Oh yes it's actually the opposite of what I thought.
Ty